### ISSUE TITLE
Multidimensional array of objects get flattened

### ISSUE THREAD

**Issue by cjwilliams24680** [apollographql/apollo-android#1160]
On our backend, we've defined a return type that contains a multidimensional array of data objects like this:

```
const Examples = gql`
    type Examples {
        grid: [[Example!]!]!
    }
`;

const Example = gql`
    type Example {
        id: ID!
        value: String!
    }
`;

const ExampleQueries = gql`
    type Query {
        examplesQuery(): Examples!
    }
`;
```

I've written a query definition like this in the graphql file of my Android project:
```
query GetExamples() {
    examplesQuery() {
        grid {
            id
            value
        }
    }
}
```

When I generate the Java code, I get `final @NotNull List<Grid> grid;` instead of the two dimensional array in the schema. When iOS generates their code using the same files, they get a two dimensional array as expected.

On Android, we're using the library version 1.0.0-alpha2.

Any help is appreciated, thanks!

**Comment by sav007** [comment]
pls try on snapshot version, I believe we fixes smth like this

**Comment by cjwilliams24680** [comment]
Yep looks like it's fixed on `1.0.1-SNAPSHOT`. Thanks!
